platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```diff
@@ -0,0 +1,205 @@
+            }
+
+            OutputStream outStream = new FileOutputStream(crlFile);
+            outStream.write(buffer);
+            SecurityLogger.audit(
+                "Copied the content of the CRl at {} to the local CRL at {}.",
+                crlLocationUrl,
+                localCrlPath);
+
+            setCrlFileLocationInPropertiesFile(localCrlPath);
+          }
+        }
+      } catch (IOException e) {
+        LOGGER.warn("Unable to copy CRL to local CRL.");
+      }
+    } else {
+      LOGGER.warn("No/bad response");
+    }
+  }
+
+  /** Sets the org.apache.ws.security.crypto.merlin.x509crl.file property */
+  @VisibleForTesting
+  void setCrlFileLocationInPropertiesFile(String localCrlPath) {
+    addProperty(issuerSignaturePropertiesLocation, localCrlPath);
+    addProperty(issuerEncryptionPropertiesLocation, localCrlPath);
+    addProperty(serverSignaturePropertiesLocation, localCrlPath);
+    addProperty(serverEncryptionPropertiesLocation, localCrlPath);
+
+    SecurityLogger.audit(
+        "Setting the {} property to {} as signature and encryption properties.",
+        CRL_PROPERTY_KEY,
+        localCrlPath);
+  }
+
+  private void addProperty(String path, String localCrlPath) {
+    Properties properties = PROPERTIES_LOADER.loadProperties(path);
+    properties.put(CRL_PROPERTY_KEY, localCrlPath);
+    try (OutputStream outputStream = new FileOutputStream(path)) {
+      properties.store(outputStream, null);
+    } catch (IOException e) {
+      LOGGER.warn("Unable to add the {} property to the property file {}.", CRL_PROPERTY_KEY, path);
+    }
+  }
+
+  /** Removes the org.apache.ws.security.crypto.merlin.x509crl.file property */
+  void removeCrlFileLocationInPropertiesFile() {
+    removeProperty(issuerSignaturePropertiesLocation);
+    removeProperty(issuerEncryptionPropertiesLocation);
+    removeProperty(serverSignaturePropertiesLocation);
+    removeProperty(serverEncryptionPropertiesLocation);
+
+    SecurityLogger.audit(
+        "Removing the {} property from signature and encryption properties.", CRL_PROPERTY_KEY);
+  }
+
+  private void removeProperty(String path) {
+    Properties properties = PROPERTIES_LOADER.loadProperties(path);
+    properties.remove(CRL_PROPERTY_KEY);
+    try (OutputStream outputStream = new FileOutputStream(path)) {
+      properties.store(outputStream, null);
+    } catch (IOException e) {
+      LOGGER.warn(
+          "Unable to remove the {} property to the property file {}.", CRL_PROPERTY_KEY, path);
+    }
+  }
+
+  public void setCrlLocationUrl(String crlLocationUrl) {
+    this.crlLocationUrl = crlLocationUrl;
+    run();
+  }
+
+  public void setCrlByUrlEnabled(boolean crlByUrlEnabled) {
+    if (this.crlByUrlEnabled && !crlByUrlEnabled) {
+      removeCrlFileLocationInPropertiesFile();
+    }
+
+    this.crlByUrlEnabled = crlByUrlEnabled;
+  }
+
+  @VisibleForTesting
+  static void setIssuerEncryptionPropertiesLocation(String issuerEncryptionPropertiesLocation) {
```

Comment:
✏️ It seems that you left the attributes at package-private level even though you were introducing these methods. If you are leaving them as package-private to read them then I see no need for these methods.